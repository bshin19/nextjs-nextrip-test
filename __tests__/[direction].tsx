import { render } from "@testing-library/react";
import Stops, { getServerSideProps } from "../pages/route/[route]/[direction]";

describe("Stops", () => {
  it("should match minimum config", () => {
    const { container } = render(<Stops />);

    expect(container).toMatchSnapshot();
  });

  it("should match with stops", () => {
    const { container } = render(
      <Stops
        stops={[
          { place_code: "901", description: "MN State Capitol" },
          { place_code: "48", description: "Mall of America" },
        ]}
      />
    );

    expect(container).toMatchSnapshot();
  });
});

describe("getServerSideProps", () => {
  it("should return the appropriate props", async () => {
    const context = { params: { route: 100, direction: 0 } };
    expect(await getServerSideProps(context as any)).toMatchSnapshot();
  });

  it('should return empty object without direction', async () => {
    const context = { params: { route: 100 } };
    expect(await getServerSideProps(context as any)).toMatchSnapshot();
  })

  it('should return empty object without route', async () => {
    const context = { params: { direction: 0 } };
    expect(await getServerSideProps(context as any)).toMatchSnapshot();
  })

  it('should return empty object with empty params', async () => {
    const context = { };
    expect(await getServerSideProps(context as any)).toMatchSnapshot();
  })
});
