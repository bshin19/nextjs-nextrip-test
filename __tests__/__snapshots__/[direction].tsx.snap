// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stops should match minimum config 1`] = `
<div>
  <h3
    class="text-center fg-secondary"
  >
    Stops along route:
  </h3>
  <ul
    class="grid flex-wrap gap-1 list-none grid-implicit-rows-even break-word hyphens"
  />
</div>
`;

exports[`Stops should match with stops 1`] = `
<div>
  <h3
    class="text-center fg-secondary"
  >
    Stops along route:
  </h3>
  <ul
    class="grid flex-wrap gap-1 list-none grid-implicit-rows-even break-word hyphens"
  >
    <li
      class="bg-primary fg-inverse border p-.25 cols-4 cols-md-2 rounded flex align-center"
    >
      <div
        class="fs-1 fs-md-2"
      >
        üöè
      </div>
      <span>
        1
        . 
        MN State Capitol
      </span>
    </li>
    <li
      class="bg-primary fg-inverse border p-.25 cols-4 cols-md-2 rounded flex align-center"
    >
      <div
        class="fs-1 fs-md-2"
      >
        üöè
      </div>
      <span>
        2
        . 
        Mall of America
      </span>
    </li>
  </ul>
</div>
`;

exports[`getServerSideProps should return empty object with empty params 1`] = `
Object {
  "props": Object {},
}
`;

exports[`getServerSideProps should return empty object without direction 1`] = `
Object {
  "props": Object {},
}
`;

exports[`getServerSideProps should return empty object without route 1`] = `
Object {
  "props": Object {},
}
`;

exports[`getServerSideProps should return the appropriate props 1`] = `
Object {
  "props": Object {
    "directions": Array [
      Object {
        "direction_id": "0",
        "direction_name": "Northbound",
      },
      Object {
        "direction_id": "1",
        "direction_name": "Southbound",
      },
    ],
    "stops": Array [
      Object {
        "description": "Minnetonka",
        "place_code": "122",
      },
      Object {
        "description": "MN Zoo",
        "place_code": "101",
      },
    ],
  },
}
`;
